services/job_service.py

```
@@ -143,162 +194,117 @@ def create_job(
+            def _delete_job():
+                db.session.delete(job)
+                return job
 
-        elif is_error(result):
-            current_app.logger.error(f"Failed to delete job {job_id}: {result.error}")
-            return error_result(result.error)
+            return self.safe_execute(_delete_job)
 
-        return error_result("Unknown error during deletion")
+        except Exception as e:
+            current_app.logger.error(f"Error deleting job {job_id}: {str(e)}", exc_info=True)
```

Ruff (0.13.1)

Use explicit conversion flag

Replace with conversion flag

(RUF010)